### PR TITLE
Removing decision stump class

### LINKED ISSUE #2205: Removing DecisionStump class from mlpack
`DecisionStump` has been marked as deprecated see #2099 . So, in mlpack 4.0.0 release we need to entirely remove `DecisionStump`  as user will use the `DecisionTree`
    class with `NoRecursion=true` or use `ID3DecisionStump` instead of using `DecisionStump`.

### PR DESCRIPTION
**Pull request by bisakhmondal**
Hello maintainers, I have tried to fix #2205 . This is my first PR as a contributor to mlpack. Any suggestion would be highly appreciated.
Description: 
1. here I have removed the decision stump module after thoroughly investigating that the package is deprecated and every package has either shifted to `tree::ID3DecisionStump` or  `DecisionTree`
class with `NoRecursion=true`.
2. Updated The CMakeLists.
3. Tests updated.
4. a bit of documentation updated.

### CODE REVIEW

**Comment by birm** [comment]
Thanks for your first PR! Just to remind you, we won't approve this until we're very close to the 4.0.0 release.

**Comment by bisakhmondal** [comment]
@birm Thanks. I understand. Not an issue.

**Comment by rcurtin** [comment]
Now that we've released the last mlpack 3.x version, we can merge this PR.  I merged master into this, just to check that everything still works. :+1:

**Comment by bisakhmondal** [comment]
Fingers crossed 🤞

File: src/mlpack/methods/adaboost/adaboost.hpp [hunk 1]
```diff
@@ -71,7 +71,8 @@ namespace adaboost {
  * @endcode
  *
  * For more information on and examples of weak learners, see
- * perceptron::Perceptron<> and decision_stump::DecisionStump<>.
+ * perceptron::Perceptron<> and 
+ tree::ID3DecisionStump.
```
Comment (rcurtin) [inline src/mlpack/methods/adaboost/adaboost.hpp:75]:
```suggestion
 * perceptron::Perceptron<> and tree::ID3DecisionStump.
```

**Review by rcurtin — COMMENTED** [review]
Hey @bisakhmondal, looks great.  Thanks for working on this and sorry it took so long to get to a point where we could merge it. :)

I think we can also remove all the files in `src/mlpack/methods/decision_stump/` and `src/mlpack/tests/decision_stump_test.cpp`.  If you'd like to do that it would be great, otherwise I can push to your branch.  :+1:

**Comment by rcurtin** [comment]
Ok---I think everything is ready here. :+1:  I won't review it though since I contributed to it too. :)

**Review by shrit — APPROVED** [review]
Everything looks good to me. :+1:

**Review by zoq — APPROVED** [review]
Looks good to me as well, great that we can replace the decision stump class.

**Comment by rcurtin** [comment]
Goodbye decision stump.  You served up well. :)

**Pull request merged**